### PR TITLE
feat: allow invalidating remote candidates

### PR DESCRIPTION
**Pull request by thomaseizinger**
The new API discussed in https://github.com/algesten/str0m/pull/496. Opening here for further discussion as a draft. Will report back with results of experiment but the passing unit test is already giving me good vibes :)

### CODE REVIEW

File: src/ice/mod.rs [hunk 1]
```diff
@@ -302,6 +303,64 @@ mod test {
...
+
+            if a1_nominated && a2_nominated {
+                break;
+            }
+
+            progress(&mut a1, &mut a2);
+        }
+
+        assert!(a1.time.duration_since(a1_time) < STUN_TIMEOUT);
+        assert!(a2.time.duration_since(a2_time) < STUN_TIMEOUT);
```
Comment (thomaseizinger) [inline src/ice/mod.rs:361]: Without the consideration of remote-candidates in the `invalidate_candidate` function, this assertion fails.

**Review by algesten — APPROVED** [review]
This looks good!

**Comment by thomaseizinger** [comment]
Prelimiary testing suggests that this works quite well! Watch this space.

**Comment by thomaseizinger** [comment]
This has been working really well for us and it is not even an API addition itself, just expanding the scope of `invalidate_candidate`.

There is an issue with candidate invalidation in general though, see https://github.com/algesten/str0m/pull/504.

**Comment by algesten** [comment]
Thanks!

**Pull request merged**